### PR TITLE
[gui] confirm keyboard dialog with enter while on edit control

### PR DESCRIPTION
**Pull request by xhaggi**
I was a bit annoyed by the fact that I could not confirm the keyboard dialog by pressing enter while stay on the edit control. You have to navigate to the confirm button. By default we map `enter` to `select` in our [keyboard.xml](https://github.com/xbmc/xbmc/blob/master/system/keymaps/keyboard.xml#L45)

This PR will address the issue.

### CODE REVIEW

File: xbmc/dialogs/GUIDialogKeyboardGeneric.cpp [hunk 1]
```diff
@@ -198,7 +198,8 @@ bool CGUIDialogKeyboardGeneric::OnAction(const CAction &action)
   bool handled = true;
   if (actionId == (KEY_VKEY | XBMCVK_BACK))
     Backspace();
-  else if (actionId == ACTION_ENTER || (m_isKeyboardNavigationMode && action.GetID() == ACTION_SELECT_ITEM))
+  else if (actionId == ACTION_ENTER ||
+           ((m_isKeyboardNavigationMode || GetFocusedControlID() == CTL_EDIT) && action.GetID() == ACTION_SELECT_ITEM))
     OnOK();
```
Comment (garbear) [inline xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: This looks much clearer split over multiple lines:

```c++
else if (actionId == ACTION_ENTER)
  OnOK();
else if (actionId == ACTION_SELECT_ITEM &&
    (m_isKeyboardNavigationMode || GetFocusedControlID() == CTL_EDIT))
  OnOK();
```

Comment (xhaggi) [reply xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: -1 this will introduce dupe code.

Comment (xhaggi) [reply xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: switched the conditions so that the actionId is checked first, like in your suggestion.

Comment (garbear) [reply xbmc/dialogs/GUIDialogKeyboardGeneric.cpp]: Looks much better, thanks

**Comment by garbear** [comment]
No more objections from my end. Thanks for doing the cleanup.

**Comment by xhaggi** [comment]
Thanks for your review 👍

**Pull request merged**